app/src/main/java/com/hover/stax/transfers/TransactionType.kt

```diff
@@ -0,0 +1,13 @@
+package com.hover.stax.transfers
+
+import timber.log.Timber
+import kotlin.properties.Delegates
+
+class TransactionType {
```

Comment:
Could you explain why this is like this?

Reply:
The issue was with how the ViewModels are initialised from the `Koin` module. If we access the `ViewModel` normally where an activity + fragment or 2 fragments share one, then a new instance of the `ViewModel` will be created for each fragment. Therefore, all variables will be reset to their original values making passing info and state difficult. To solve this, we use access the `ViewModel` in either classes as a `SharedViewModel`. You can see this in action in the `BalancesFragment`.

If two fragments share a `ViewModel` but are not hosted in the same activity, then the method above fails. The only way I could find to send the transaction type was to use a delegates.